coins/monero/src/wallet/send/multisig.rs

```diff
@@ -153,22 +145,7 @@ impl SignableTransaction {
       clsags.push(AlgorithmMachine::new(clsag, offset));
     }
 
-    // Select decoys
-    // Ideally, this would be done post entropy, instead of now, yet doing so would require sign
-    // to be async which isn't preferable. This should be suitably competent though
-    // While this inability means we can immediately create the input, moving it out of the
-    // Arc RwLock, keeping it within an Arc RwLock keeps our options flexible
-    let decoys = Decoys::select(
-      // Using a seeded RNG with a specific height, committed to above, should make these decoys
-      // committed to. They'll also be committed to later via the TX message as a whole
-      &mut ChaCha20Rng::from_seed(transcript.rng_seed(b"decoys")),
-      rpc,
-      self.protocol.ring_len(),
-      height,
-      &self.inputs,
-    )
-    .await
-    .map_err(TransactionError::RpcError)?;
+    let decoys = self.decoys.clone();
```

Comment:
You shouldn't clone decoys out of self when self is in the Machine. Just access them via `.signable`.